src/chainrules.jl

```diff
@@ -41,3 +41,15 @@ function ChainRulesCore.rrule(
     end
     return C, mul_pullback
 end
+
+function ChainRulesCore.rrule(
+    ::typeof(dot), A::AbstractSparseMatrix{T},B::AbstractSparseMatrix{T}
+) where T
+    r = dot(A, B)
```

Comment:
May be `typeof(.)` is better since `typeof(*)` is used rather than `typeof(mul)`?

Reply:
Unsure what you mean here. Can you elaborate on this more?

The idea here is to pirate the `rrule` for `AbstractSparseMatrix`, so if mean doing `rrule(::typeof(*), x::T, y::T) where T<:Number` you are doing it in a scalar level. And that would be more controversial because we are not certain how it implies to hijeck the ChainRules AD system.

Reply:
I just mean that `x ⋅ y` is a synonym for `dot( x,  y )`. Maybe in most cases  ` ⋅` is used more often than `dot` in AD system?

Reply:
It's identical https://github.com/JuliaLang/julia/blob/f916b94298fa15407b139bbdd2d6079a65ae6ed8/stdlib/LinearAlgebra/src/LinearAlgebra.jl#L447

I prefer the name `dot` here as it's more "explicit". `⋅` can sometimes be confusing if you consider `.` and `⊙`, `⊗`, `⊡` (https://github.com/JuliaMath/TensorCore.jl/blob/master/src/TensorCore.jl)